pkg/kubelet/kubelet.go

```diff
@@ -2612,7 +2612,17 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 // updateContainerResourceAllocation updates AllocatedResources values
 // (for cpu & memory) from checkpoint store
 func (kl *Kubelet) updateContainerResourceAllocation(pod *v1.Pod) {
-	for _, c := range pod.Spec.Containers {
+	containers := pod.Spec.Containers
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+		for _, c := range pod.Spec.InitContainers {
+			if kubetypes.IsRestartableInitContainer(&c) {
+				containers = append(containers, c)
+			}
+		}
+	}
+
+	for _, c := range containers {
```

Comment:
Does this actually work? It's modifying the resource requests on the container, but none of container, resources, or requests are pointers...

I'd rather extract the loop body to an anonymous function and call it while looping over InitContainers and Containers separately.

Reply:
Yes actually it was a bug since the assignment of arrays in Golang is copy by value. I have made the changes as per your suggestion.